test/e2e/apimachinery/flowcontrol.go

```diff
@@ -247,6 +252,224 @@ var _ = SIGDescribe("API priority and fairness", func() {
+
+		ginkgo.By("watching")
+		framework.Logf("starting watch")
+		fsWatch, err := client.Watch(ctx, metav1.ListOptions{ResourceVersion: list.ResourceVersion, LabelSelector: "my-fs-label=" + labelValue})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("patching")
+		fsPatched, err := client.Patch(ctx, fsCreated.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
+		framework.ExpectNoError(err)
+		gomega.Expect(fsPatched.Annotations).To(gomega.HaveKeyWithValue("patched", "true"), "patched object should have the applied annotation")
+
+		ginkgo.By("updating")
+		fsToUpdate := fsPatched.DeepCopy()
+		fsToUpdate.Annotations["updated"] = "true"
+		fsUpdated, err := client.Update(ctx, fsToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		gomega.Expect(fsUpdated.Annotations).To(gomega.HaveKeyWithValue("updated", "true"), "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAnnotation := false; !sawAnnotation; {
+			select {
+			case evt, ok := <-fsWatch.ResultChan():
+				if !ok {
+					framework.Fail("watch channel should not close")
+				}
+				gomega.Expect(evt.Type).To(gomega.Equal(watch.Modified))
+				fsWatched, isFS := evt.Object.(*flowcontrol.FlowSchema)
+				if !isFS {
+					framework.Failf("expected an object of type: %T, but got %T", &flowcontrol.FlowSchema{}, evt.Object)
+				}
+				if fsWatched.Annotations["patched"] == "true" {
+					sawAnnotation = true
+					fsWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", fsWatched.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
+		ginkgo.By("getting /status")
+		resource := flowcontrol.SchemeGroupVersion.WithResource("flowschemas")
+		fsStatusRead, err := f.DynamicClient.Resource(resource).Get(ctx, fsCreated.Name, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err)
+		gomega.Expect(fsStatusRead.GetObjectKind().GroupVersionKind()).To(gomega.Equal(flowcontrol.SchemeGroupVersion.WithKind("FlowSchema")))
+		gomega.Expect(fsStatusRead.GetUID()).To(gomega.Equal(fsCreated.UID))
+
+		ginkgo.By("patching /status")
+		fsStatusPatched, err := client.Patch(ctx, fsCreated.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patchedstatus":"true"}},"status":{"conditions":[{"type":"PatchStatusFailed","status":"False","reason":"e2e"}]}}`), metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+		condition := apihelpers.GetFlowSchemaConditionByType(fsStatusPatched, flowcontrol.FlowSchemaConditionType("PatchStatusFailed"))
+		gomega.Expect(condition).NotTo(gomega.BeNil())
+		gomega.Expect(fsStatusPatched.Annotations).To(gomega.HaveKeyWithValue("patchedstatus", "true"), "patched object should have the applied annotation")
```

Comment:
I would not expect this line to work ... we should verify we did *not* update the annotation via status

Reply:
yes, this test is no longer attempting to update annotation when doing UpdateStatus